app/controllers/TaxCodeChangeController.scala

```diff
@@ -61,10 +71,32 @@ trait TaxCodeChangeController extends TaiBaseController
       implicit request =>
         if(taxCodeChangeEnabled) {
           ServiceCheckLite.personDetailsCheck {
-            Future.successful(Ok(views.html.taxCodeChange.yourTaxFreeAmount()))
+            val nino = Nino(user.getNino)
+
+            val employmentNameFuture = employmentService.employmentNames(nino, TaxYear())
+            val taxCodeChangeFuture = taxCodeChangeService.taxCodeHistory(nino)
+
+            for {
+              employmentNames <- employmentNameFuture
+              taxCodeChange <- taxCodeChangeFuture
+              codingComponents <- codingComponentService.taxFreeAmountComponents(nino, TaxYear())
+              companyCarBenefits <- companyCarService.companyCarOnCodingComponents(nino, codingComponents)
+
+            } yield {
+              (taxCodeChange) match {
+                case (TaiSuccessResponseWithPayload(taxCodeHistory: TaxCodeHistory)) => {
+                  taxCodeHistory.taxCodeRecord match {
```

Comment:
We have a case here for this option to be `None` which we don't handle, so we could run into a match error. Instead we can handle that and simplify the nested matching by matching in the first case on 

`TaiSuccessResponseWithPayload(TaxCodeHistory(_, Some(taxCodeRecords: Seq[TaxCodeRecord])))`

instead of 

```
case (TaiSuccessResponseWithPayload(taxCodeHistory: TaxCodeHistory)) => {
                  taxCodeHistory.taxCodeRecord match {
	                    case Some(taxCodeRecords) =>
```